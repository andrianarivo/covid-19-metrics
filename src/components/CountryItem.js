import PropTypes from 'prop-types';
import moment from 'moment';
import styled from 'styled-components';

export default function CountryItem(props) {
  const { country, active, updated } = props;
  return (
    <Country>
      <td>{country}</td>
      <td>
        <ul>
          <li>
            {active}
          </li>
          <li>{`${moment(new Date(updated)).fromNow()}`}</li>
        </ul>
      </td>
    </Country>
  );
}

CountryItem.propTypes = {
  country: PropTypes.string.isRequired,
  active: PropTypes.number.isRequired,
  updated: PropTypes.number.isRequired,
};

const Country = styled.tr`
  width: 100%;
  height: 5rem;

  td:nth-child(1) {
    padding-left: 1rem;
  }

  ul {
    li {
      list-style: none;
    }
  }

  &:nth-child(n) {
    background-color: #dc4781;
  }

  &:nth-child(2n) {
    background-color: #cf4378;
  }
`;
