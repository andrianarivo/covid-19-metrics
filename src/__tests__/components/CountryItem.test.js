import { create } from 'react-test-renderer';
import CountryItem from '../../components/CountryItem';

describe('Testing CountryItem component', () => {
  test('renders correctly', () => {
    const tree = create(<CountryItem country="Dummy Country" active={0} updated={0} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
