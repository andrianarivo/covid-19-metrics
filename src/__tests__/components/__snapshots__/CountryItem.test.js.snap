// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CountryItem component renders correctly 1`] = `
<tr
  className="sc-jzuGiw YszEF"
>
  <td>
    Dummy Country
  </td>
  <td>
    <ul>
      <li>
        0
      </li>
      <li>
        53 years ago
      </li>
    </ul>
  </td>
</tr>
`;
